### ISSUE TITLE
Parts of libcompiler-rt are compiled as WX

### ISSUE THREAD

**Issue by djc** [rust-lang/rust#34482]
The following parts have WX sections (that is, memory that is writable and executable at the same time):

```
 * !WX --- --- usr/lib64/rust-1.9.0/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler-rt.a:chkstk.o
 * !WX --- --- usr/lib64/rust-1.9.0/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler-rt.a:chkstk2.o
 * !WX --- --- usr/lib64/rust-1.9.0/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler-rt.a:floatundidf.o
 * !WX --- --- usr/lib64/rust-1.9.0/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler-rt.a:floatundisf.o
 * !WX --- --- usr/lib64/rust-1.9.0/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler-rt.a:floatundixf.o
```

From a security point of view, this is probably bad.

**Comment by trofi** [comment]
llikley proper upstream is llvm: 
                   https://github.com/llvm-mirror/compiler-rt/blob/master/lib/builtins/x86_64/chkstk2.S

I suspect it lacks GNU stack marking. Example marking from libffi: 
                   https://github.com/libffi/libffi/blob/master/src/x86/sysv.S#L1038

**Comment by Aatch** [comment]
Hard to tell if it's an oversight or not from the code. There's a macro that outputs the stack marking if needed, which is used in other files. Either way, this is an upstream issue as it's in compiler-rt.

**Comment by Amanieu** [comment]
This seems to have been fixed upstream very recently: https://github.com/llvm-mirror/compiler-rt/commit/4c71a477ac0bd7b19c72d7625861f1f18c15b278

**Comment by alexcrichton** [comment]
If this has been fixed upstream, then that's great! The steps for fixing this would then look like:

* Check out the most recent branch of compiler-rt that we're using, which I believe is rust-lang/compiler-rt at the branch rust-2016-06-16
* Cherry-pick the relevant commit to the branch
* Send a PR to the rust-lang/compiler-rt repo on the rust-2016-06-16 branch
* Once merged, send a PR to rust-lang/rust updating the compiler-rt repo
* Ideally write a test as well, but may be somewhat difficult to do that for this...

I'm gonna tag this as E-easy and I'm certainly willing to mentor, if anyone needs any help just let me know!

**Comment by apasel422** [comment]
Closed in #34647.

**Comment by djc** [comment]
It seems like this has regressed in 1.14.0:

```
 * !WX --- --- usr/lib64/rust-1.14.0/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-f5a209a9.rlib:chkstk.o
 * !WX --- --- usr/lib64/rust-1.14.0/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-f5a209a9.rlib:chkstk2.o
 * !WX --- --- usr/lib64/rust-1.14.0/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-f5a209a9.rlib:floatundidf.o
 * !WX --- --- usr/lib64/rust-1.14.0/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-f5a209a9.rlib:floatundisf.o
 * !WX --- --- usr/lib64/rust-1.14.0/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-f5a209a9.rlib:floatundixf.o
```
